src/main/java/org/prebid/server/bidder/huaweiads/HuaweiUtilRequest.java

```diff
@@ -0,0 +1,474 @@
+                    .width(device.getW())
+                    .language(device.getLanguage())
+                    .pxratio(device.getPxratio())
+                    .belongCountry(country)
+                    .localeCountry(country)
+                    .ip(device.getIp())
+                    .gaid(gaid)
+                    .oaid(oaid)
+                    .imei(imei)
+                    .clientTime(clientTime)
+                    .isTrackingEnabled(resolveIsTrackingEnabled(dnt, oaid))
+                    .isGaidTrackingEnabled(resolveIsTrackingEnabled(dnt, gaid)).build();
+        }
+        return HuaweiDevice.builder().build();
+    }
+
+    private static String resolveId(List<String> listId) {
+        if (listId == null) {
+            return null;
+        }
+        return !listId.isEmpty() ? listId.get(0) : null;
+    }
+
+    private static ExtUserDataDeviceIdHuaweiAds resolveDeviceData(BidRequest request, JacksonMapper mapper) {
+        final User user = request.getUser();
+        final ExtUserDataHuaweiAds extUserDataHuaweiAds;
+        if (user == null) {
+            return null;
+        }
+        ExtUser userExt = user.getExt();
+        if (userExt == null) {
+            return null;
+        }
+
+        try {
+            extUserDataHuaweiAds = mapper.mapper().readValue(userExt.getData().toString(),
+                    ExtUserDataHuaweiAds.class);
+        } catch (JsonProcessingException e) {
+            throw new PreBidException("Unmarshal: BidRequest.user.ext -> extUserDataHuaweiAds failed");
+        }
+        return extUserDataHuaweiAds.getData();
+    }
+
+    private static String resolveCountry(BidRequest bidRequest) {
+        Geo geo;
+        String country;
+
+        Device device = bidRequest.getDevice();
+        geo = device != null ? device.getGeo() : null;
+        country = geo != null ? geo.getCountry() : null;
+        if (!StringUtils.isBlank(country)) {
+            return convertCountryCode(country);
+        }
+
+        User user = bidRequest.getUser();
+        geo = user != null ? user.getGeo() : null;
+        country = geo != null ? geo.getCountry() : null;
+        if (!StringUtils.isBlank(country)) {
+            return convertCountryCode(country);
+        }
+
+        return DEFAULT_COUNTRY_NAME;
+    }
+
+    private static String convertCountryCode(String country) {
+        Map<String, String> mapCountryCodeAlpha3ToAlpha2 = new HashMap<>();
+        mapCountryCodeAlpha3ToAlpha2.put("CHL", "CL");
+        mapCountryCodeAlpha3ToAlpha2.put("CHN", "CN");
+        mapCountryCodeAlpha3ToAlpha2.put("ARE", "AE");
+        String countryPresentInMap = mapCountryCodeAlpha3ToAlpha2.keySet().stream()
+                .filter(element -> element.equals(country))
+                .findFirst()
+                .orElse(null);
+
+        if (countryPresentInMap != null) {
+            return mapCountryCodeAlpha3ToAlpha2.get(countryPresentInMap);
+        }
```

Comment:
just mapCountryCodeAlpha3ToAlpha2.get(country) will be enough in pair with Map.of();